### PR TITLE
booster-http: update docs for new minimal trustless mode

### PR DESCRIPTION
**Pull request by rvagg**
I don't know how well this is going to work as a PR, I know GitBook has moved away from GitHub as the primary interface and this could get janky, but here we go.

The diff is fairly large, I've done a comprehensive overhaul of the HTTP docs, reusing a lot of it and extending other bits.

* The main shift is in describing how `booster-http` does "trustless" retrievals, and explaining a bit about that.
* All the trusted stuff has been removed and replaced with a section about how `bifrost-gateway` can be used to set that up.
* `serving-files-with-booster-http.md` has been replaced with `advanced-http-configuration.md` that covers config options, `bifrost-gateway` trusted setup and reverse proxy setup.
* NGINX instructions have been extended a bit with some additional bits (compression, TLS, caching), mainly pointers to docs.

Too dense? I won't be offended if this is trimmed, snipped, whatever. Just trying to move things forward.

### CODE REVIEW

File: retrieving-data-from-filecoin/http-retrieval.md [hunk 1]
```diff
@@ -2,55 +2,65 @@
...
 
-To return the CAR file for a given CID, you can pass an `Accept` header with the `application/vnd.ipld.car;` format. This can be useful for retrieving the raw, unpadded data of a deal.
+To download a CAR file containing all the blocks linked under a given root CID, you can either pass an `Accept: application/vnd.ipld.car` header, or a `?format=car` query parameter to the `/ipfs/` endpoint. Likewise, to download raw single IPLD block bytes, you can either pass an `Accept: application/vnd.ipld.raw` header, or a `?format=raw` query parameter to the `/ipfs/` endpoint.
+
+The `/ipfs/` endpoint must be followed by a CID and one of the indicators above, for example:
 
-{% code overflow="wrap" %}
 ```
-curl -H "Accept:application/vnd.ipld.car;" http://{SP's http retrieval URL}/ipfs/bafySomePayloadCID -o bafySomePayloadCID.car
+curl 'http://{SP's HTTP retrieval URL}/ipfs/bafySomePayloadCID?format=car' -o bafySomePayloadCID.car
```
Comment (LexLuthr) [inline retrieving-data-from-filecoin/http-retrieval.md]:
```suggestion
curl 'http://{SP's HTTP retrieval URL}/ipfs/bafySomePayloadCID?format=car' -o bafySomePayloadCID.car
curl 'http://{SP's HTTP retrieval URL}/ipfs/bafySomeBlockCID?format=raw' -o bafySomeBlockCID
```

File: retrieving-data-from-filecoin/http-retrieval.md [hunk 2]
```diff
@@ -2,55 +2,65 @@
...
 ```
```
Comment (LexLuthr) [inline retrieving-data-from-filecoin/http-retrieval.md:57]:
```suggestion
```

OR

```
curl -H "Accept:application/vnd.ipld.car;" http://{SP's http retrieval URL}/ipfs/bafySomePayloadCID -o bafySomePayloadCID.car
curl -H "Accept:application/vnd.ipld.raw;" http://{SP's http retrieval URL}/ipfs/bafySomeBlockCID -o bafySomeBlockCID

```

```

File: retrieving-data-from-filecoin/advanced-http-configuration.md [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+   --tracing-endpoint value                                 the endpoint for the tracing exporter (default: "http://tempo:14268/api/traces")
+   --api-filter-endpoint value                              the endpoint to use for fetching a remote retrieval configuration for requests
+   --api-filter-auth value                                  value to pass in the authorization header when sending a request to the API filter endpoint (e.g. 'Basic ~base64 encoded user/pass~'
+   --badbits-denylists value [ --badbits-denylists value ]  the endpoints for fetching one or more custom BadBits list instead of the default one at https://badbits.dwebops.pub/denylist.json (default: "https://badbits.dwebops.pub/denylist.json")
+   --help, -h                                               show help
+```
+
+### booster-http, boostd and lotus
+
+`booster-http` must have network, or localhost access to a full `lotus` node, a `lotus-miner` and a `boostd` instance. The following options are required:
```
Comment (LexLuthr) [inline retrieving-data-from-filecoin/advanced-http-configuration.md]:
```suggestion
`booster-http` must have network, or localhost access to a full `lotus` node, a `lotus-miner` and a `LID` instance. The following options are required:
```

File: retrieving-data-from-filecoin/advanced-http-configuration.md [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+
+* `--api-lid` - the `boostd` Local Index Directory (_LID_)
```
Comment (LexLuthr) [inline retrieving-data-from-filecoin/advanced-http-configuration.md]:
```suggestion
* `--api-lid` - the Local Index Directory (_LID_) service endpoint
```

File: retrieving-data-from-filecoin/advanced-http-configuration.md [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+                auth_basic_user_file /etc/nginx/ipfs-gateway.conf.d/.htpasswd;
+
+                limit_req zone=client_ip_10rs;
+                limit_req_status 429;
+                proxy_pass http://127.0.0.1:7777;
+        }
+}
+```
+
+1. Click the refresh button in your browser on any path under `/ipfs/` more than once per second you will see a 429 error page.
```
Comment (LexLuthr) [inline retrieving-data-from-filecoin/advanced-http-configuration.md]:
```suggestion
4. Click the refresh button in your browser on any path under `/ipfs/` more than once per second you will see a 429 error page.
```

**Review by LexLuthr — APPROVED** [review]
Few minor changes. Otherwise, looks good.
Please DO NOT MERGE till we have a RC or Stable release with the relevant code changes.

**Comment by rvagg** [comment]
feedback addressed, and I also tinkered with all of the shell script blocks, making them properly `sh` so they syntax highlight nicely and replacing this kind of thing: `{SP's HTTP retrieval URL}` with this kind of thing: `{booster-http exposed url}` (partly to get around the problematic `'` character in there).

**Pull request merged**